classes/Langchecker/Utils.php

```diff
@@ -124,18 +124,32 @@ public static function secureText($origin, $isarray = true)
     }
 
     /**
-     * Highlight Python variables in string
+     * Get Python variables from a string. Checking for
+     * %(var)s, %s, and %% ('escaped' percentage sign)
      *
      * @param array $origin Original string
      *
-     * @return string String withon Python variables marked with <em>
+     * @return array Python variables
      */
-    public static function highlightPythonVar($origin)
+    public static function getPythonVariables($origin)
     {
         $origin = htmlspecialchars($origin);
         $regex = '#%(\([a-z0-9._-]+\)s|[s%])#';
         preg_match_all($regex, $origin, $matches);
-        foreach ($matches[0] as $python_var) {
+
+        return $matches[0];
+    }
+
+    /**
+     * Highlight Python variables in string
+     *
+     * @param array $origin Original string
+     *
+     * @return string String with Python variables marked with <em>
+     */
+    public static function highlightPythonVar($origin)
+    {
+        foreach (self::getPythonVariables($origin) as $python_var) {
```

Comment:
We need to keep `$origin = htmlspecialchars($origin);` here, otherwise the HTML code will be interpreted when displaying the wrong string.

On the other hand, we should be able to remove the call in getPythonVariables (I thought I commented on that one, but I'd need to check the old comments).